relay-server/src/services/processor/ourlog.rs

```diff
@@ -135,19 +139,81 @@ pub fn process(
     Ok(())
 }
 
+/// Config needed to normalize a standalone log.
+#[derive(Clone, Debug)]
+struct NormalizeOurLogConfig {
+    /// The user agent parsed from the request.
+    user_agent: Option<String>,
+    /// Client hints parsed from the request.
+    client_hints: ClientHints<String>,
+}
+
+impl NormalizeOurLogConfig {
+    fn new(managed_envelope: &ManagedEnvelope) -> Self {
+        Self {
+            user_agent: managed_envelope
+                .envelope()
+                .meta()
+                .user_agent()
+                .map(String::from),
+            client_hints: managed_envelope.meta().client_hints().clone(),
+        }
+    }
+}
+
 #[cfg(feature = "processing")]
-fn normalize(annotated_log: &mut Annotated<OurLog>) -> Result<(), ProcessingError> {
+fn normalize(
+    annotated_log: &mut Annotated<OurLog>,
+    config: NormalizeOurLogConfig,
```

Comment:
This can take a reference:

```suggestion
    config: &NormalizeOurLogConfig,
```